public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -659,18 +659,34 @@ export class LokiDatasource
 
   modifyQuery(query: LokiQuery, action: QueryFixAction): LokiQuery {
     let expression = query.expr ?? '';
+    // NB: Usually the labelKeys should be fetched and cached in the datasource,
+    // but there might be some edge cases where this wouldn't be the case.
+    // However the changed would make this method `async`.
+    const allLabels = this.languageProvider.getLabelKeys();
     switch (action.type) {
       case 'ADD_FILTER': {
         if (action.options?.key && action.options?.value) {
           const value = escapeLabelValueInSelector(action.options.value);
-          expression = addLabelToQuery(expression, action.options.key, '=', value);
+          expression = addLabelToQuery(
+            expression,
+            action.options.key,
+            '=',
+            value,
+            allLabels.includes(action.options.key) === false
```

Comment:
I think that this condition should have a check for the parser, because otherwise it breaks behaviour for all parsed labels, as it forces `forceAsLabelFilter` for parsed labels as well. 

https://github.com/grafana/grafana/assets/30407135/7d87d19f-6a7b-49a5-acfc-de42725147d8

Reply:
Solved with 0241cabafffc08eda0522b3b8f051c41ad45f668. WDYT?